### PR TITLE
External ids

### PR DESCRIPTION
**Pull request by lowendavid**
## Type of change

- [ ] Feature

## Description of change

The buildRequests function now can retreive external ids information.
Unit test is setup to make sure the information is well stored and read.

The following values need to be added following this pattern.
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  },
  userId: {
      britepoolid: '1111',
      criteoId: '1111',
      digitrustid: { data: { id: 'DTID', keyv: 4, privacy: { optout: false }, producer: 'ABC', version: 2 } },
      id5id: '1111',
      idl_env: '1111',
      lipbid: '1111',
      parrableid: 'eidVersion.encryptionKeyReference.encryptedValue',
      pubcid: '1111',
      tdid: '1111',
      netId: 'fH5A3n2O8_CZZyPoJVD-eabc6ECb7jhxCicsds7qSg',
    }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

## Other information
contribution by lowen@smartadserver.com

### CODE REVIEW

File: modules/smartadserverBidAdapter.js [hunk 1]
```diff
@@ -82,6 +82,27 @@ export const spec = {
...
+          if (source === 'digitrustid' && obj && typeof obj === 'object' && obj.data && obj.data.id) {
+            foundId = obj.data.id;
+          } else if (obj && typeof obj === 'string') {
+            foundId = obj;
+          }
+          if (foundId !== '') {
+            if (payload.eids === undefined) {
+              payload.eids = [];
+            }
+            payload.eids.push({
```
Comment (smenzer) [inline modules/smartadserverBidAdapter.js:99]: why not just use `userIdsAsEids` instead, that way the identity providers can handle the formatting for you without all the extra code above?

Comment (lowendavid) [reply modules/smartadserverBidAdapter.js]: Excellent point, now i use createEidsArray found in eid.js, works a treat.

**Comment by smenzer** [comment]
you should also make sure to update your prebid.org docs with support for userIds, like this:

`userIds: britepoolId, criteo, digitrust, id5Id, identityLink, liveIntentId, netId, parrableId, pubCommonId, unifiedId`

**Comment by lowendavid** [comment]
> please add docs, otherwise LGTM

this PR https://github.com/prebid/prebid.github.io/pull/2062 contains changes to documentation, i think it should cover what you are asking for.
Otherwise let me know what more i can do or if i misunderstood your request 🙂
Thanks

**Comment by smenzer** [comment]
Since @harpere approved pending docs, and docs were added, I'm going ahead with merging this.

**Pull request merged**